src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -978,8 +978,12 @@ public OwnerDTO updateOwner(@Verify(Owner.class) String key,
                 .updateOwnerContentAccess(owner, contentAccessModeList, contentAccessMode);
         }
 
-        owner = ownerCurator.merge(owner);
-        ownerCurator.flush();
+        // ContentAccessManager does its own merge on owner, if updates are related to content access
+        // this is redundant.
+        if (!updateContentAccess) {
```

Comment:
I'm hesitant to okay this, since we intentionally left it off prior to avoid logical coupling to the CAM for merging the product data. With the added comment it's probably fine, as future maintainers would have a hint as to why the merge isn't always done; but in such a case we should reuse the ```updateContentAccess``` check we've already done directly above this block and drop it in as an "else" block.

Reply:
I was initially thinking of putting everything inside of else block.
Done.